server/src/routes/clientRouter.js

```diff
@@ -3,7 +3,7 @@ import getClientIndexPath from "../config/getClientIndexPath.js";
 
 const router = new express.Router();
 
-const clientRoutes = ["/", "/user-sessions/new", "/users/new", "/:categoryId"];
+const clientRoutes = ["/", "/user-sessions/new", "/users/new", "/categories/id", "/products/:id"];
```

Comment:
Careful here! You need to update the `/categories/:id` path to have a colon for the dynamic route